### PR TITLE
[risk=low][RW-11847] Easy prerequisites for the spring-boot upgrade

### PR DESCRIPTION
**Pull request by jmthibault79**
First, bump spring-boot to the latest 2.x (this is not the major upgrade - that will be to 3.x)

Then made these prerequisite upgrades:
* mysql-connector
* spring-security
* jackson

Tested locally with no obvious issues.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: api/build.gradle [hunk 1]
```diff
@@ -19,8 +19,9 @@ buildscript {
         OKHTTP_VERSION = '2.7.5'
         OPENCENSUS_VERSION = '0.31.1'
         SPRINGFOX_VERSION = '3.0.0'
-        SPRING_BOOT_VERSION = '2.7.4'
+        SPRING_BOOT_VERSION = '2.7.18'
         SPRING_FRAMEWORK_VERSION = '5.3.23'
+        SPRING_SECURITY_VERSION = '5.8.9'
```
Comment (jmthibault79) [inline api/build.gradle:24]: We will need to migrate to 6.x soon.  5.8.x is a transitional release to make that easier: https://docs.spring.io/spring-security/reference/5.8/migration/index.html

File: api/build.gradle [hunk 2]
```diff
@@ -425,7 +426,7 @@ dependencies {
     implementation 'io.swagger:swagger-annotations:1.6.7'
     implementation "javax.inject:javax.inject:1"
     implementation "org.liquibase:liquibase-core:$project.ext.LIQUIBASE_VERSION"
-    implementation 'mysql:mysql-connector-java:8.0.30'
+    implementation 'com.mysql:mysql-connector-j:8.3.0'
```
Comment (jmthibault79) [inline api/build.gradle:429]: note the new package name

File: api/build.gradle [hunk 3]
```diff
@@ -483,8 +484,9 @@ dependencies {
     }
 
     implementation "org.springframework.retry:spring-retry"
-    implementation "org.springframework.security:spring-security-core"
-    implementation "org.springframework.security:spring-security-web"
+    implementation "org.springframework.security:spring-security-core:$project.ext.SPRING_SECURITY_VERSION"
+    implementation "org.springframework.security:spring-security-crypto:$project.ext.SPRING_SECURITY_VERSION"
```
Comment (jmthibault79) [inline api/build.gradle:488]: when I didn't call crypto one out explicitly, it chose an older version

File: api/db-cdr/build.gradle [hunk 4]
```diff
@@ -17,7 +17,7 @@ dependencies {
     liquibaseRuntime 'org.liquibase:liquibase-core:4.16.1'
     liquibaseRuntime 'org.liquibase:liquibase-groovy-dsl:3.0.2'
     liquibaseRuntime 'info.picocli:picocli:4.6.1'
-    liquibaseRuntime 'mysql:mysql-connector-java:5.1.37'
+    liquibaseRuntime 'com.mysql:mysql-connector-j:8.3.0'
```
Comment (jmthibault79) [inline api/db-cdr/build.gradle:20]: tested via liquibase migrations

**Review by freemabd — APPROVED** [review]
LGTM

**Pull request merged**